internal/hardware/hardware.go

```diff
@@ -14,16 +13,17 @@ func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
 		GhwChrootRoot: "/",
 	})
 
-	// Instead of copying field-by-field marshal to JSON and unmarshal to other struct
-	inventoryJson, err := json.Marshal(inv)
-	if err != nil {
-		return nil, err
-	}
 	hardwareInfo := models.HardwareInfo{}
-	err = json.Unmarshal(inventoryJson, &hardwareInfo)
-	if err != nil {
-		return nil, err
+
+	hardwareInfo.CPU = &models.CPU{
+		Architecture: inv.CPU.Architecture,
+		ModelName:    inv.CPU.ModelName,
+		Flags:        []string{},
 	}
 
+	hardwareInfo.Hostname = inv.Hostname
+
+	hardwareInfo.SystemVendor = (*models.SystemVendor)(inv.SystemVendor)
```

Comment:
GPU info and flags details are missing (I understand they were removed in https://issues.redhat.com/browse/ECOPROJECT-327 for clarity and not as a requirement of removal) @masayag - is that correct?

Reply:
I understood that it should be removed. But let's wait for @masayag cause he was in the meeting.

Reply:
@masayag

Reply:
yes, they should be removed.